public/agg_types/comparing.js

```diff
@@ -32,6 +37,30 @@ function getDate(date, offset) {
   return date.clone().subtract(offset.value, offset.unit).toISOString();
 }
 
+/**
+ * Rounds up `customComparing.to` to the end of the day if needed.
+ * Returns the current time (`moment()`) if the field is missing.
+ *
+ * @param { from, to } customComparing
+ */
+function handleCustomDate(customComparing) {
+  // If customComparing is missing,
+  //  returns the object as if the fields are empty.
+  if (!customComparing) return { from: moment(), to: moment() };
+
+  // Checks if `customComparing.to` is using day format.
+  //  If so, rounds it to the end of the day
+  const isEndDateUsingTime = customComparing.to && customComparing.to.includes(':');
```

Comment:
What if the input date is an hour, like `2018/04/21 17`? Will it work?

Reply:
Unless you explicitly specify this format, the `YYYY/MM/DD HH` pattern is invalid for `moment`, so you'll get an error message in form level, before you can run the query (take a look at the [validate_date_math.js](https://github.com/elastic/kibana/blob/master/src/ui/public/directives/validate_date_math.js) directive used in this field). I know this is not the best solution to check if the string contains time, but I couldn't find another easy way to accomplish that (I wish `moment` had this kind of validation)